lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -50,7 +50,7 @@ def get(self):
         Return server configuration information required by web clients
         including the Pbench dashboard UI. This includes:
 
-        openid-connect: A JSON object containing the OpenID Connect parameters
+        openid: A JSON object containing the OpenID Connect parameters
```

Comment:
The code here is still written as if the `openid` section is still optional, which is really isn't. I'm almost tempted to suggest removing the `try/except` or at least changing the `pass` into an `APIInternalError` ... but I also don't like failing `GET /endpoints` and it "shouldn't be possible". 😦

Reply:
Hmm. The way it is structured right now looks like the Pbench server can not start without the `openid` section configured with Keycloak server URL, however, the endpoints API will not return an exception if the `opeid` section is missing. I am not sure if there ever be a situation where we are able to start the server container but later the config file got corrupted and now the endpoints API is not returning the `openid` config options. But yeah as you said do we want endpoints API to fail on such occasions or do we just not pass the `openid` config value?

Reply:
Yeah, the complication is that we've moved into "the server won't start" territory, which means that errors "can't" happen in `GET /endpoints`. So one view is that we just grab the config settings we need, *knowing* they're present, and if some bizarre internal error occurs, it's handled as a bizarre internal error in the `APIBase` dispatcher without complicating/confusing the code here.

Or ... we can be paranoid, deal with the possibility of errors here, explicitly, and ... so something. I'm not really sure what that would be. Returning without `openid` data is pointless, as the client might legitimately require that data and we know it *has to exist*. Raising an internal error is no more helpful.

Reply:
Yeah, we'll teach @npalaska not to pull on threads like this one -- you never know what might unravel from it!  🤣 

Anyway, as I understand it, the call to `self.server_config.get()` doesn't reopen the config file -- it works on a cached copy in memory from the initial access -- so I don't think we have to worry about having the config file switched or corrupted out from under us.

And, we really should aspire to checking the whole configuration on startup and dying as soon as possible if something is bad or missing...just so that when we access it later (like here) we don't have to consider the possibility that what we need isn't there.

Given that, I think Nikhil should tear out the `try` block for the `NoOptionError` and `NoSectionError` exceptions, since those should never happen.  (And, if they do happen, Flask will handle them....)

Reply:
I agree I dont think the try-except block here is serving any purpose now.